src/kokkos/plugin-PixelTriplets/PixelDoubletsConstants.h

```diff
@@ -0,0 +1,86 @@
+#ifndef RecoPixelVertexing_PixelTriplets_plugins_PixelDoubletsConstants_h
+#define RecoPixelVertexing_PixelTriplets_plugins_PixelDoubletsConstants_h
+
+#include "CAConstants.h"
+
+namespace PixelDoubletsConstants {
+  constexpr int nPairs = 13 + 2 + 4;
+  static_assert(nPairs <= CAConstants::maxNumberOfLayerPairs());
+
+  // constants via functor member variables
+  class LayerPairs {
+  private:
+    const uint8_t layerPairs[2 * nPairs] = {
+        0, 1, 0, 4, 0, 7,              // BPIX1 (3)
+        1, 2, 1, 4, 1, 7,              // BPIX2 (5)
+        4, 5, 7, 8,                    // FPIX1 (8)
+        2, 3, 2, 4, 2, 7, 5, 6, 8, 9,  // BPIX3 & FPIX2 (13)
+        0, 2, 1, 3,                    // Jumping Barrel (15)
+        0, 5, 0, 8,                    // Jumping Forward (BPIX1,FPIX2)
+        4, 6, 7, 9                     // Jumping Forward (19)
+    };
+
+  public:
+    KOKKOS_INLINE_FUNCTION uint8_t operator[](int i) const { return layerPairs[i]; };
+  };
```

Comment:
The idea was actually to define the constants as members of the functor itself. I.e. implement `getDoubletsFromHisto` itself as a class with the function call operator called directly by the `parallel_for`.

Reply:
yeah, I found it's weird also. Is the latest update correct?

Reply:
Thanks, it looks like what I was after.

Reply:
Thanks